src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Hex2Dec.txt

```diff
@@ -0,0 +1,57 @@
+﻿>> Hex2Dec("0")
+0
+
+>> Hex2Dec(Blank())
+#Error(Kind=InvalidArgument)
+
+// Positive Integer Parameters
+>> Hex2Dec("A5")
+165
+
+>> Hex2Dec("C8D")
+3213
+
+>> Hex2Dec("B2D4")
+45780
+
+>> Hex2Dec("7FFFFFFFFF")
+549755813887
+
+// Negative Integer Parameters
+>> Hex2Dec("FFFFFFFFFF")
+-1
+
+>> Hex2Dec("FFFFFFFF5B")
+-165
+
+>> Hex2Dec("FFFFFFF373")
+-3213
+
+>> Hex2Dec("8000000002")
+-549755813886
+
+>> Hex2Dec("8000000001")
+-549755813887
+
+>> Hex2Dec("FFFFFF4D2C")
+-45780
+
+
+
+// ******** NULL and ERROR PARAMETERS ********
+// More than 10 characters
+>> Hex2Dec("FFFFFFFFFFF")
+#Error(Kind=Numeric)
+
+// characters no in 0-9 and A-Z range
+>> Hex2Dec("ZZZZ")
+#Error(Kind=InvalidArgument)
+
+>> Hex2Dec(["A","12","FFFF"])
+[10,18,65535]
+
+>> Hex2Dec("43j")
+#Error(Kind=InvalidArgument)
+
+>> Hex2Dec(["3","E","I","B"])
+[3,14,Microsoft.PowerFx.Types.ErrorValue,11]
```

Comment:
>] [](http://example.com/codeflow?start=43&length=1)

More tests:

- Parameters with different casings: Hex2Dec("FfFf") ==> 65535
- Error inputs: Hex2Dec(If(1/0<2,"value")) ==> Div0 error
- Errors / blank values in tables: Hex2Dec(Table({a:"F"},If(1/0<2,{a:"b"}),{a:"d"},{a:""},If(1<0,{a:"e"}),{a:"a"})) ==> Table({Value:15},Error,{Value:13},{Value:0},Blank(),{Value:10})